src/controller.ts

```diff
@@ -252,19 +261,46 @@ export default function (
       },
     );
 
-    fastify.post<{ 
+    fastify.get<{ Querystring: { code: string; provider: string } }>(
+      '/is-unique-code-available',
+      {
+        schema: {
+          querystring: {
+            type: 'object',
+            properties: { code: { type: 'string' }, provider: { type: 'string' } },
+          },
+        },
+        config: {
+          rateLimit: {
+            max: 5,
+            timeWindow: '1 minute',
+          },
+        },
+      },
+      async (req, res) => {
+        const { code, provider } = req.query;
+        try {
+          await licenseCodesService.isLicenseCodeAvailable(code, provider);
+          return res.status(200).send({ message: 'Code is available' });
+        } catch (err) {
```

Comment:
Let's make it more clear: 

```
if (err instanceof LicenseCodeAlreadyAppliedError || err instanceof InvalidLicenseCodeError) {
   return res.status(404).send({ message: err.message });
}

req.log.error(`[LICENSE/CHECK/ERROR]: ${err.message}. STACK ${err.stack || 'NO STACK'}`);
return rep.status(500).send({ message: 'Internal Server Error' });
```

Reply:
Think that if you have any error there, you want to know it. You cannot assume that all the errors are coming from what you expect to find. At that moment, you will need the logs to exist and to be easy to find to troubleshoot any issue @masterprog-cmd.